### PR TITLE
[R4R] Upgrade to 1.10.3

### PR DESCRIPTION
**Pull request by yutianwu**
## Description

Upgrade to go-ethereum 1.10.3.

### Snapshots
Snapshots are an acceleration data structure on top of the Ethereum state, that allows reading accounts and contract storage significantly faster. The snapshot feature reduces the cost of accessing an account from O(logN) to O(1).

Though snapshots have enormous benefits, there are certain costs to them:
A snapshot is a redundant copy of the raw Ethereum state already contained in the leaves of the Merkle Patricia trie so it requires an additional disk overhead.
Since nobody has snapshots constructed in the network yet, nodes will initially need to bear the cost of iterating the state trie and creating the initial snapshot themselves. This might take a day to a week but you only need to do it once in the lifetime of your node.

Snapshot feature is enabled by default, you can disable it via `--snapshot=false`.

#### Snap sync
Now you have two different ways to synchronize the BSC network: full sync and fast sync. Full sync operated by downloading the entire chain and executing all transactions; fast sync placed an initial trust in a recent-ish block, and directly downloaded the state associated with it (after which it switched to block execution like full sync).

For fast sync, it needs to download the trie nodes one by one. So it will take time to download all the nodes if there are millions of nodes. And for the serving peers, they need to traverse all the nodes and it also takes time.

Snap sync is designed to solve the fast sync problems.  The core idea is instead of downloading the trie node-by-node, snap sync downloads the contiguous chunks of useful state data, and reconstructs the Merkle trie locally.

You can manually enable snap sync via `--syncmode=snap`.  Note: usually it will need one or two weeks until there is enough client upgrade to the version with snap enabled, you may use snap sync after that. 

#### Offline pruning
If you have snapshots enabled and fully generated, Geth can use those as an acceleration structure to relatively quickly determine which trie nodes should be kept and which should be deleted. Pruning trie nodes based on snapshots does have the drawback that the chain may not progress during pruning. This means that you need to stop Geth, prune its database and then restart it.

Execution time wise, pruning takes a few hours (greatly depends on your disk speed and accumulated junk), one third of which is indexing recent trie nodes from snapshots, one third deleting stale trie nodes and the last third compacting the database to reclaim freed up space. At the end of the process, your disk usage should approximately be the same as if you did a fresh sync.

To prune your database, please run `geth snapshot prune-state`.

Note:  **You need to upgrade the client and run for a while before prune the data**

### Transaction unindexing

Geth no longer keeps transaction inclusion info for all transactions, and instead limits the storage of inclusion records to one year. For application developers, this change means that very old transactions can no longer be accessed by hash.
If you would like todisable this behavior and keep inclusion information for all historical transactions, you can re-enable indexing using the `--txlookuplimit=0` command-line flag.

### Database changes
There is an incompatible database change: adds a prefix for contract code in order to separate the codes and trie nodes.

So if you use the version of geth, you will not be able to rollback to the older version geth.

### Geth command changes

+ --rpc -> --http - Enable the HTTP-RPC server
+ --rpcaddr -> --http.addr - HTTP-RPC server listening interface
+ --rpcport -> --http.port - HTTP-RPC server listening port
+ --rpccorsdomain -> --http.corsdomain - Domain from which to accept requests
+ --rpcvhosts -> --http.vhosts - Virtual hostnames from which to accept requests
+ --rpcapi -> --http.api - API’s offered over the HTTP-RPC interface
+ --wsaddr -> --ws.addr - WS-RPC server listening interface
+ --wsport -> --ws.port - WS-RPC server listening port
+ --wsorigins -> --ws.origins - Origins from which to accept websockets requests
+ --wsapi -> --ws.api - API’s offered over the WS-RPC interface
+ --gpoblocks -> --gpo.blocks - Number of blocks to check for gas prices
+ --gpopercentile -> --gpo.percentile - Percentile of recent txs to use as gas suggestion
+ --graphql.addr -> --graphql - Enable GraphQL on the HTTP-RPC server
+ --graphql.port -> --graphql - Enable GraphQL on the HTTP-RPC server
+ --pprofport -> --pprof.port - Profiler HTTP server listening port
+ --pprofaddr -> --pprof.addr - Profiler HTTP server listening interface
+ --memprofilerate -> --pprof.memprofilerate - Turn on memory profiling with the given rate
+ --blockprofilerate -> --pprof.blockprofilerate - Turn on block profiling with the given rate
+ --cpuprofile -> --pprof.cpuprofile - Write CPU profile to the given file

For more details about the command changes, you can refer to: v1.10.0.

### Rpc changes

**BREAKING CHANGE**:  Non-EIP155 transactions (i.e. transactions which are not replay-protected) are now rejected by the RPC API.  **You can disable this restriction using the `--rpc.allow-unprotected-txs` command-line flag**.
Since there are txs signed by old clients, we need to add this flag for all our nodes.

#### Response changes in `eth_estimateGas`, `eth_call`, `eth_sendRawTransaction`.
The response will now add a message field in addition to the data field if these RPCs cause the EVM to execute a REVERT operation. For example:
```
{
   "jsonrpc":"2.0",
   "id":1,
   "error":{
      "code":3,
      "message":"execution reverted: num_ is \u003c= 10",
      "data":"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d6e756d5f206973203c3d20313000000000000000000000000000000000000000"
   }
}
```

#### Response changes in `eth_getTransactionByBlockHashAndIndex`, `eth_getTransactionByBlockNumberAndIndex`, `eth_getTransactionByHash`.
The response now adds a type field to denote the transaction type. For example:
```
{
   "jsonrpc":"2.0",
   "id":1,
   "result":{
      "blockHash":"0x812619b0a7eb58de4f4c4fcf0b497567705f9182c2ad102745fa5ed87b3d96b5",
      "blockNumber":"0x95dc",
      "from":"0xef7368f755f5f551a602971b1f792b2737bc6e16",
      "gas":"0x2dc6c0",
      "gasPrice":"0x430e23400",
      "hash":"0xabbed20a397ff811fddf2b3f559e6291c1d14c6456bd1eca5fde82169e283857",
      "input":"0x29e99f07000000000000000000000000000000000000000000000000000000000000000c",
      "nonce":"0x55731a",
      "to":"0x6ac00c8b5514496aaae72cc2e6b8d4c29ee7d4ba",
      "transactionIndex":"0x0",
      "value":"0x0",
      "type":"0x0",
      "v":"0x1b",
      "r":"0x6e8c8af7e3266f0216b9a796eddd736a4f9bcdbfd6a47c49f566c187174d3268",
      "s":"0x5bf210f56d04126e0864643cb06670abe46fbf935446438490c2ff9ce6921adf"
   }
}
```
### GraphQL changes

The GraphQL API is no longer available separately from the JSON-RPC HTTP server. If you want GraphQL, you need to enable it using the `--http --graphql` flag combination. The `--graphql.port` and `--graphql.addr` flags are no longer available.

#### Remove whisper config.

```
[Shh]
MaxMessageSize = 1048576
RestrictConnectionBetweenLightClients = true
```

#### Remove GraphQL config.

```
GraphQLPort
GraphQLVirtualHosts
```

#### Rename `DiscoveryURLs` to `EthDiscoveryURLs`.

For more details about config changes, you can refer to go-ethereum v1.9.19, v1.9.21, v1.10.0

### Preflight checks

- [x] build passed (`make build`)
- [ ] tests passed (`make test`)
- [ ] manual transaction test passed

### CODE REVIEW

**Comment by maoueh** [comment]
@yutianwu Is it ready to be tested? I would give it a go to test if helps with recent syncing issues we are seeing on the network.

**Comment by guagualvcha** [comment]
> @yutianwu Is it ready to be tested? I would give it a go to test if helps with recent syncing issues we are seeing on the network.

Hi, it is not well tested, please do not use it in production.

**Comment by maoueh** [comment]
@guagualvcha  But is it tested enough to be "working"? Would like to gather metrics about this new version.

**Comment by DebugFuture** [comment]
Does bsc have a plan to merge latest code from geth?

**Comment by maoueh** [comment]
@DebugFuture this is already a merge of the most up-to-date released version ...

Are you asking if there is a plan to deploy it on the network?

**Comment by heliosVII-tech** [comment]
If bsc upgraded, I feel like it’s a big change and all the old nodes would have to be told to upgrade. What is the planned timeline for releasing it 2 the network?

**Comment by maoueh** [comment]
As long as the hard fork elements are not activated, both old and new version could co-exists.

And if hard fork are planned to be activated, I would assume that just like for Ethereum, it will be announced well in advanced.

**Comment by guagualvcha** [comment]
> @guagualvcha But is it tested enough to be "working"? Would like to gather metrics about this new version.

Yes, it is tested enough to be "working", we have used this client to syn to the latest block from genesis. You can use it for test and only for test so far, thanks.

**Comment by DebugFuture** [comment]
@maoueh yes, I'm asking when will the new version be deployed to main net, especially for the validator nodes. Thanks.

**Comment by maoueh** [comment]
@guagualvcha Great thank you, I'll give some performance numbers related to syncing between this version and the current release number later on.

**Comment by mmitech** [comment]
adding the chainId is not sufficient, the node will need to have --rpc.allow-unprotected-txs for your app to be able to send transactions.

Also, if your app relies on call results then you need to change the way you handle the results since reverts are now thrown as exceptions

**Comment by mmitech** [comment]
it is probably better to wait a couple more days for geth maintenance release (v1.10.3) that will be released in a couple of days, it has a more optimized snap sync code.

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Build Geth in a stock Go builder container
 FROM golang:1.15-alpine as builder
 
 RUN apk add --no-cache make gcc musl-dev linux-headers git bash
```
Comment (guagualvcha) [inline Dockerfile:4]: This docker file is written when the golang 1.15 base image is not ready,, please refer to https://github.com/ethereum/go-ethereum/blob/master/Dockerfile, to use golang 1.16

File: README.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ The goal of Binance Smart Chain is to bring programmability and interoperability
 
 Binance Smart Chain starts its development based on go-ethereum fork. So you may see many toolings, binaries and also docs are based on Ethereum ones, such as the name “geth”.
 
+[![API Reference](
+https://camo.githubusercontent.com/915b7be44ada53c290eb157634330494ebe3e30a/68747470733a2f2f676f646f632e6f72672f6769746875622e636f6d2f676f6c616e672f6764646f3f7374617475732e737667
+)](https://pkg.go.dev/github.com/ethereum/go-ethereum?tab=doc)
+[![Go Report Card](https://goreportcard.com/badge/github.com/ethereum/go-ethereum)](https://goreportcard.com/report/github.com/ethereum/go-ethereum)
+[![Travis](https://travis-ci.com/ethereum/go-ethereum.svg?branch=master)](https://travis-ci.com/ethereum/go-ethereum)
+[![Discord](https://img.shields.io/badge/discord-join%20chat-blue.svg)](https://discord.gg/nthXNEv)
```
Comment (guagualvcha) [inline README.md:12]: Please change the Discord link to BSC one

File: README.md [hunk 3]
```diff
@@ -76,13 +83,25 @@ directory.
...
-|  **`geth`**   | Main Binance Smart Chain client binary. It is the entry point into the BSC network (main-, test- or private net), capable of running as a full node (default), archive node (retaining all historical state) or a light node (retrieving data live). It has the same and more RPC and other interface as go-ethereum and can be used by other processes as a gateway into the BSC network via JSON RPC endpoints exposed on top of HTTP, WebSocket and/or IPC transports. `geth --help` and the [CLI Wiki page](https://github.com/ethereum/go-ethereum/wiki/Command-Line-Options) for command line options.          |
-|   `abigen`,`bootnode`,`evm`, `gethrpctest`,`rlpdump`,`puppeth`     | **These binaries are exactly the same as the binaries built in go-ethereum repo.**|
+<<<<<<< HEAD
```
Comment (guagualvcha) [inline README.md:86]: Confict not resolved?

File: README.md [hunk 4]
```diff
@@ -76,13 +83,25 @@ directory.
...
+=======
+|  **`geth`**   | Our main Ethereum CLI client. It is the entry point into the Ethereum network (main-, test- or private net), capable of running as a full node (default), archive node (retaining all historical state) or a light node (retrieving data live). It can be used by other processes as a gateway into the Ethereum network via JSON RPC endpoints exposed on top of HTTP, WebSocket and/or IPC transports. `geth --help` and the [CLI page](https://geth.ethereum.org/docs/interface/command-line-options) for command line options.          |
+|   `clef`    | Stand-alone signing tool, which can be used as a backend signer for `geth`.  |
+|   `devp2p`    | Utilities to interact with nodes on the networking layer, without running a full blockchain. |
```
Comment (guagualvcha) [inline README.md:91]: remove this

File: README.md [hunk 5]
```diff
@@ -112,12 +146,72 @@ use separate accounts for play-money and real-money. Unless you manually move
 accounts, `geth` will by default correctly separate the two networks and will not make any
 accounts available between them.*
 
+### Full node on the Rinkeby test network
```
Comment (guagualvcha) [inline README.md:149]: This part is not needed.

File: README.md [hunk 6]
```diff
@@ -112,12 +146,72 @@ use separate accounts for play-money and real-money. Unless you manually move
...
+To get an idea how the file should look like you can use the `dumpconfig` subcommand to
+export your existing configuration:
+
+```shell
+$ geth --your-favourite-flags dumpconfig
+```
+
+*Note: This works only with `geth` v1.6.0 and above.*
+
+#### Docker quick start
```
Comment (guagualvcha) [inline README.md:189]: This part is not needed

File: README.md [hunk 7]
```diff
@@ -112,12 +146,72 @@ use separate accounts for play-money and real-money. Unless you manually move
...
+```
+
+To get an idea how the file should look like you can use the `dumpconfig` subcommand to
+export your existing configuration:
+
+```shell
+$ geth --your-favourite-flags dumpconfig
+```
+
+*Note: This works only with `geth` v1.6.0 and above.*
```
Comment (guagualvcha) [inline README.md:187]: why v1.6.0 here?

File: README.md [hunk 8]
```diff
@@ -112,12 +146,72 @@ use separate accounts for play-money and real-money. Unless you manually move
...
+and/or hosts. By default, `geth` binds to the local interface and RPC endpoints is not
+accessible from the outside.
+
 ### Programmatically interfacing `geth` nodes
 
 As a developer, sooner rather than later you'll want to start interacting with `geth` and the
-Binance Smart Chain network via your own programs and not manually through the console. To aid
-this, `geth` has built-in support for a JSON-RPC based APIs ([standard APIs](https://github.com/ethereum/wiki/wiki/JSON-RPC)
-and [`geth` specific APIs](https://github.com/ethereum/go-ethereum/wiki/Management-APIs)).
+Ethereum network via your own programs and not manually through the console. To aid
```
Comment (guagualvcha) [inline README.md:212]: Binance Smart Chain

File: README.md [hunk 9]
```diff
@@ -153,6 +247,148 @@ BSC nodes with exposed APIs! Further, all browser tabs can access locally
 running web servers, so malicious web pages could try to subvert locally available
 APIs!**
 
+### Operating a private network
```
Comment (guagualvcha) [inline README.md:250]: No need to add this part.

File: README.md [hunk 10]
```diff
@@ -153,6 +247,148 @@ BSC nodes with exposed APIs! Further, all browser tabs can access locally
...
+```shell
+$ geth <usual-flags> --mine --miner.threads=1 --miner.etherbase=0x0000000000000000000000000000000000000000
+```
+
+Which will start mining blocks and transactions on a single CPU thread, crediting all
+proceedings to the account specified by `--miner.etherbase`. You can further tune the mining
+by changing the default gas limit blocks converge to (`--miner.targetgaslimit`) and the price
+transactions are accepted at (`--miner.gasprice`).
+
+## Contribution
```
Comment (guagualvcha) [inline README.md:366]: Please change go-ethereum, into BSC

File: SECURITY.md [hunk 11]
```diff
@@ -2,31 +2,29 @@
 
 ## Supported Versions
 
-Please see Releases. We recommend to use the most recent released version.  
+Please see [Releases](https://github.com/ethereum/go-ethereum/releases). We recommend using the [most recently released version](https://github.com/ethereum/go-ethereum/releases/latest).
```
Comment (guagualvcha) [inline SECURITY.md:5]: The link is wrong

File: SECURITY.md [hunk 12]
```diff
@@ -2,31 +2,29 @@
...
 | `clef` | 20180914 | [pdf](https://github.com/ethereum/go-ethereum/blob/master/docs/audits/2018-09-14_Clef-audit_NCC.pdf) |
 
-
-
 ## Reporting a Vulnerability
 
 **Please do not file a public ticket** mentioning the vulnerability.
 
-To find out how to disclose a vulnerability in Ethereum visit [https://bounty.ethereum.org](https://bounty.ethereum.org) or email bounty@ethereum.org.
+To find out how to disclose a vulnerability in Ethereum visit [https://bounty.ethereum.org](https://bounty.ethereum.org) or email bounty@ethereum.org. Please read the [disclosure page](https://github.com/ethereum/go-ethereum/security/advisories?state=published) for more information about publically disclosed security vulnerabilities.
```
Comment (guagualvcha) [inline SECURITY.md:20]: Please change it into BSC bounty

File: consensus/parlia/parlia.go [hunk 13]
```diff
@@ -6,6 +6,7 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"github.com/ethereum/go-ethereum/trie"
```
Comment (guagualvcha) [inline consensus/parlia/parlia.go:9]: import order

**Comment by guagualvcha** [comment]
Please refine the description of the PR, and also change the version to v1.8.0

**Comment by spec42** [comment]
Hi, my database is 560G beacause of the bug fixexd in 1.0.7-hf1. Will the prune command be able to shrink this too? Or should i sync from scratch?

**Review by guagualvcha — APPROVED** [review]

**Review by HaoyangLiu — APPROVED** [review]

**Comment by jmouvill** [comment]
Hi, how/when do we know when validators will use this release ? I suppose it will be made as a requirement at some point to be elected ? thanks

**Comment by tota999** [comment]
> ## Description
> Upgrade to go-ethereum 1.10.3.
> 
> ## Changes
> ### Snapshots
> Snapshots are an acceleration data structure on top of the Ethereum state, that allows reading accounts and contract storage significantly faster. The snapshot feature reduces the cost of accessing an account from O(logN) to O(1).
> 
> Though snapshots have enormous benefits, there are certain costs to them:
> A snapshot is a redundant copy of the raw Ethereum state already contained in the leaves of the Merkle Patricia trie so it requires an additional disk overhead.
> Since nobody has snapshots constructed in the network yet, nodes will initially need to bear the cost of iterating the state trie and creating the initial snapshot themselves. This might take a day to a week but you only need to do it once in the lifetime of your node.
> 
> Snapshot feature is enabled by default, you can disable it via `--snapshot=false`.
> 
> #### Snap sync
> Now you have two different ways to synchronize the BSC network: full sync and fast sync. Full sync operated by downloading the entire chain and executing all transactions; fast sync placed an initial trust in a recent-ish block, and directly downloaded the state associated with it (after which it switched to block execution like full sync).
> 
> For fast sync, it needs to download the trie nodes one by one. So it will take time to download all the nodes if there are millions of nodes. And for the serving peers, they need to traverse all the nodes and it also takes time.
> 
> Snap sync is designed to solve the fast sync problems. The core idea is instead of downloading the trie node-by-node, snap sync downloads the contiguous chunks of useful state data, and reconstructs the Merkle trie locally.
> 
> You can manually enable snap sync via `--syncmode=snap`. Note: usually it will need one or two weeks until there is enough client upgrade to the version with snap enabled, you may use snap sync after that.
> 
> #### Offline pruning
> If you have snapshots enabled and fully generated, Geth can use those as an acceleration structure to relatively quickly determine which trie nodes should be kept and which should be deleted. Pruning trie nodes based on snapshots does have the drawback that the chain may not progress during pruning. This means that you need to stop Geth, prune its database and then restart it.
> 
> Execution time wise, pruning takes a few hours (greatly depends on your disk speed and accumulated junk), one third of which is indexing recent trie nodes from snapshots, one third deleting stale trie nodes and the last third compacting the database to reclaim freed up space. At the end of the process, your disk usage should approximately be the same as if you did a fresh sync.
> 
> To prune your database, please run `geth snapshot prune-state`.
> 
> Note: **You need to upgrade the client and run for a while before prune the data**
> 
> ### Transaction unindexing
> Geth no longer keeps transaction inclusion info for all transactions, and instead limits the storage of inclusion records to one year. For application developers, this change means that very old transactions can no longer be accessed by hash.
> If you would like todisable this behavior and keep inclusion information for all historical transactions, you can re-enable indexing using the `--txlookuplimit=0` command-line flag.
> 
> ### Database changes
> There is an incompatible database change: adds a prefix for contract code in order to separate the codes and trie nodes.
> 
> So if you use the version of geth, you will not be able to rollback to the older version geth.
> 
> ### Geth command changes
> * --rpc -> --http - Enable the HTTP-RPC server
> * --rpcaddr -> --http.addr - HTTP-RPC server listening interface
> * --rpcport -> --http.port - HTTP-RPC server listening port
> * --rpccorsdomain -> --http.corsdomain - Domain from which to accept requests
> * --rpcvhosts -> --http.vhosts - Virtual hostnames from which to accept requests
> * --rpcapi -> --http.api - API’s offered over the HTTP-RPC interface
> * --wsaddr -> --ws.addr - WS-RPC server listening interface
> * --wsport -> --ws.port - WS-RPC server listening port
> * --wsorigins -> --ws.origins - Origins from which to accept websockets requests
> * --wsapi -> --ws.api - API’s offered over the WS-RPC interface
> * --gpoblocks -> --gpo.blocks - Number of blocks to check for gas prices
> * --gpopercentile -> --gpo.percentile - Percentile of recent txs to use as gas suggestion
> * --graphql.addr -> --graphql - Enable GraphQL on the HTTP-RPC server
> * --graphql.port -> --graphql - Enable GraphQL on the HTTP-RPC server
> * --pprofport -> --pprof.port - Profiler HTTP server listening port
> * --pprofaddr -> --pprof.addr - Profiler HTTP server listening interface
> * --memprofilerate -> --pprof.memprofilerate - Turn on memory profiling with the given rate
> * --blockprofilerate -> --pprof.blockprofilerate - Turn on block profiling with the given rate
> * --cpuprofile -> --pprof.cpuprofile - Write CPU profile to the given file
> 
> For more details about the command changes, you can refer to: v1.10.0.
> 
> ### Rpc changes
> **BREAKING CHANGE**: Non-EIP155 transactions (i.e. transactions which are not replay-protected) are now rejected by the RPC API. **You can disable this restriction using the `--rpc.allow-unprotected-txs` command-line flag**.
> Since there are txs signed by old clients, we need to add this flag for all our nodes.
> 
> #### Response changes in `eth_estimateGas`, `eth_call`, `eth_sendRawTransaction`.
> The response will now add a message field in addition to the data field if these RPCs cause the EVM to execute a REVERT operation. For example:
> 
> ```
> {
>    "jsonrpc":"2.0",
>    "id":1,
>    "error":{
>       "code":3,
>       "message":"execution reverted: num_ is \u003c= 10",
>       "data":"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d6e756d5f206973203c3d20313000000000000000000000000000000000000000"
>    }
> }
> ```
> 
> #### Response changes in `eth_getTransactionByBlockHashAndIndex`, `eth_getTransactionByBlockNumberAndIndex`, `eth_getTransactionByHash`.
> The response now adds a type field to denote the transaction type. For example:
> 
> ```
> {
>    "jsonrpc":"2.0",
>    "id":1,
>    "result":{
>       "blockHash":"0x812619b0a7eb58de4f4c4fcf0b497567705f9182c2ad102745fa5ed87b3d96b5",
>       "blockNumber":"0x95dc",
>       "from":"0xef7368f755f5f551a602971b1f792b2737bc6e16",
>       "gas":"0x2dc6c0",
>       "gasPrice":"0x430e23400",
>       "hash":"0xabbed20a397ff811fddf2b3f559e6291c1d14c6456bd1eca5fde82169e283857",
>       "input":"0x29e99f07000000000000000000000000000000000000000000000000000000000000000c",
>       "nonce":"0x55731a",
>       "to":"0x6ac00c8b5514496aaae72cc2e6b8d4c29ee7d4ba",
>       "transactionIndex":"0x0",
>       "value":"0x0",
>       "type":"0x0",
>       "v":"0x1b",
>       "r":"0x6e8c8af7e3266f0216b9a796eddd736a4f9bcdbfd6a47c49f566c187174d3268",
>       "s":"0x5bf210f56d04126e0864643cb06670abe46fbf935446438490c2ff9ce6921adf"
>    }
> }
> ```
> 
> ### GraphQL changes
> The GraphQL API is no longer available separately from the JSON-RPC HTTP server. If you want GraphQL, you need to enable it using the `--http --graphql` flag combination. The `--graphql.port` and `--graphql.addr` flags are no longer available.
> 
> ### Config changes
> #### Remove whisper config.
> ```
> [Shh]
> MaxMessageSize = 1048576
> RestrictConnectionBetweenLightClients = true
> ```
> 
> #### Remove GraphQL config.
> ```
> GraphQLPort
> GraphQLVirtualHosts
> ```
> 
> #### Rename `DiscoveryURLs` to `EthDiscoveryURLs`.
> For more details about config changes, you can refer to go-ethereum v1.9.19, v1.9.21, v1.10.0
> 
> ### Preflight checks
> * [x]  build passed (`make build`)
> * [ ]  tests passed (`make test`)
> * [ ]  manual transaction test passed

**Comment by nilampatel-engineer** [comment]
Hi, running bsc full node with --gcmode = archive, syncmode=full, chain has stopped syncing with running out of space on device, if I add **geth snapshot prune-state.** it will prune database means will I loose complete historical data? means will not b able to see any chain history? what storage size is recommended? I have given 1TB and I ran out of space

**Comment by gituser** [comment]
@nilampatel-engineer for archive node `--gcmode=archive` you will need 15TB+ or maybe more.

You don't need archive node in order to get historical data unless you need to know exact balance at certain block.

**Pull request merged**